### PR TITLE
Handle NoneType self.mons from a pickle

### LINKED ISSUE #157: len(self.mons) not compatible with catalogs from proseco < 5.0
```
In [19]: acas = pickle.load(gzip.open('/data/mpcrit1/mplogs/2021/JUL1921/oflsb/output/JUL1921B_prose
    ...: co.pkl.gz'))                                                                               

In [20]: acar = acas[24524].get_review_table()                                                      

In [21]: acar.run_aca_review()                                                                      
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-21-445df580523f> in <module>
----> 1 acar.run_aca_review()

/proj/sot/ska3/flight/lib/python3.8/site-packages/sparkles/core.py in run_aca_review(self, make_html, report_dir, report_level, roll_level, roll_args, raise_exc)
    545 
    546         # Do aca review checks and update acas[0] in place
--> 547         exc = run_aca_review(load_name=f'Obsid {self.obsid}', acars=acars, make_html=make_html,
    548                              report_dir=report_dir, report_level=report_level,
    549                              roll_level=roll_level, roll_args=roll_args,

/proj/sot/ska3/flight/lib/python3.8/site-packages/sparkles/core.py in run_aca_review(load_name, acars, make_html, report_dir, report_level, roll_level, roll_args, loud, obsids, open_html, context, raise_exc)
    154     """
    155     try:
--> 156         _run_aca_review(load_name=load_name, acars=acars, make_html=make_html,
    157                         report_dir=report_dir, report_level=report_level,
    158                         roll_level=roll_level, roll_args=roll_args,

/proj/sot/ska3/flight/lib/python3.8/site-packages/sparkles/core.py in _run_aca_review(load_name, acars, make_html, report_dir, report_level, roll_level, roll_args, loud, obsids, open_html, context)
    208 
    209         aca.set_stars_and_mask()  # Not stored in pickle, need manual restoration
--> 210         aca.check_catalog()
    211 
    212         # Find roll options if requested

/proj/sot/ska3/flight/lib/python3.8/site-packages/sparkles/core.py in check_catalog(self)
    857         self.check_guide_geometry()
    858         self.check_acq_p2()
--> 859         self.check_guide_count()
    860         self.check_fid_count()
    861         self.check_include_exclude()

/proj/sot/ska3/flight/lib/python3.8/site-packages/sparkles/core.py in check_guide_count(self)
   1049         # Caution for any "unusual" guide star request
   1050         typical_n_guide = 5 if self.is_OR else 8
-> 1051         if self.n_guide + len(self.mons) != typical_n_guide:
   1052             msg = f'{obs_type} with {n_guide} guides requested but {typical_n_guide} is typical'
   1053             self.add_message('caution', msg)

TypeError: object of type 'NoneType' has no len()

```

### PR DESCRIPTION
**Pull request by jeanconn**
## Description

Handle NoneType self.mons from an old proseco pickle

## Testing

- [x] Passes unit tests linux
- [x] Functional testing - not sure if we want to add an old pickle file to the test suite.  I confirmed that with this code change I did not see the error seen in #157 .

Fixes #157

### CODE REVIEW

**Comment by jeanconn** [comment]
Another option to work around this is just to pass the call_args to (new) proseco and regenerate the catalogs, but that seems not in the spirit of things.

**Review by taldcroft — APPROVED** [review]

**Comment by taldcroft** [comment]
Agreed we should add a regression test that includes a stored pickle from proseco < 5.0 in order to avoid surprises like this. So why don't you do that as part of this PR.

**Comment by jeanconn** [comment]
Do you want to make an old pickle file or just set a new baseline with a "current" proseco pickle?  And then what paces do we need to put the code through for due diligence (run review, roll options?)?

**Comment by jeanconn** [comment]
OK.  I went with adding a single-obsid 4.12.1 proseco pkl file and added a test that uses some work-alike code to read the pickle, run review, and check for messages.

File: sparkles/tests/test_review.py [hunk 1]
```diff
@@ -507,3 +508,19 @@ def test_review_with_mon_star():
...
+
+
+def test_review_from_pickle():
+    """
+    Test that run_aca_review works from a gzip pickle file.
+    """
+
+    # This is similar to the loading from core.get_acas_from_pickle
+    # but that method does not allow full path specification.
+    filename = Path(__file__).parent / Path('data') / 'obsid.pkl.gz'
```
Comment (taldcroft) [inline sparkles/tests/test_review.py:520]: The `Path()` around `data` is not needed (the `.parent` gives a Path object).

Comment (taldcroft) [reply sparkles/tests/test_review.py:520]: Can you give the test file a more descriptive name such as `proseco_4.11.2_catalog.pkl.gz`?

**Review by taldcroft — APPROVED** [review]

**Pull request merged**